src/calls.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+use ink_core::env::EnvTypes;
+use parity_codec::{
+    Decode,
+    Encode,
+    Input,
+    Output,
+};
+use core::convert::TryInto;
+
+#[cfg_attr(feature = "std", derive(Debug, Clone, PartialEq, Eq))]
+pub enum Address<T: EnvTypes, AccountIndex> {
+    Id(T::AccountId),
+    Index(AccountIndex),
+}
+
+/// Returns `b` if `b` is greater than `a` and otherwise `None`.
+fn need_more_than<T: PartialOrd>(a: T, b: T) -> Option<T> {
```

Comment:
Please rename this.

Reply:
Copied from substrate, but couldn't think of a better name. Have gone for `greater_than_or_none`